src/main/scala/zio/intellij/inspections/simplifications/SimplifyForeachInspection.scala

```diff
@@ -0,0 +1,96 @@
+package zio.intellij.inspections.simplifications
+
+import org.jetbrains.plugins.scala.codeInspection.collections._
+import org.jetbrains.plugins.scala.lang.psi.api.expr.{ScExpression, ScFor}
+import zio.intellij.inspections._
+import zio.intellij.inspections.zioMethods.`.*>`
+
+class SimplifyForeachInspection
+    extends ZInspection(
+      ForeachForCompSimplificationType,
+      ForeachParForCompSimplificationType,
+      ForeachParNForCompSimplificationType,
+      ForeachChainSimplificationType,
+      ForeachParChainSimplificationType,
+      ForeachParNChainSimplificationType
+    )
+
+sealed abstract class BaseForeachSimplificationType(methodName: String) extends SimplificationType {
+
+  override def hint: String = s"Replace with ZIO.$methodName"
+
+  protected def replacement(expr: ScExpression, iterable: ScExpression, func: ScExpression): Simplification =
+    replace(expr).withText(s"ZIO.$methodName(${iterable.getText})(${func.getText})").highlightFrom(expr)
+}
+
+sealed abstract class BaseForeachParNSimplificationType extends SimplificationType {
+
+  private val methodName = "foreachParN_"
+
+  override def hint: String = s"Replace with ZIO.$methodName"
+
+  def replacement(expr: ScExpression, n: ScExpression, iterable: ScExpression, func: ScExpression): Simplification =
+    replace(expr).withText(s"ZIO.$methodName(${n.getText})(${iterable.getText})(${func.getText})").highlightFrom(expr)
+}
+
+sealed abstract class BaseForeachForCompSimplificationType(
+  methodName: String,
+  methodExtractor: ZIOCurried2StaticMemberReference
+) extends BaseForeachSimplificationType(methodName) {
+
+  override def getSimplification(expr: ScExpression): Option[Simplification] =
+    expr match {
+      case ScFor(enumerators, _) =>
+        enumerators.generators.collectFirst {
+          case `_ <- x`(expr @ methodExtractor(iterable, func)) => replacement(expr, iterable, func)
+        }
+      case _ => None
+    }
+}
```

Comment:
You should probably override another similar method:
```suggestion
  override def getSimplifications(expr: ScExpression): Seq[Simplification] =
    expr match {
      case ScFor(enumerators, _) =>
        enumerators.generators.collect {
          case `_ <- x`(expr @ methodExtractor(iterable, func)) => replacement(expr, iterable, func)
        }
      case _ => Nil
    }
}
```
so that all `_ <- foreach...` would be highlighted.

Reply:
I didn't notice this method, thanks!
Fixed.